core/src/main/java/com/linecorp/armeria/server/RouteBindingBuilders.java

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.linecorp.armeria.common.HttpMethod.CONNECT;
+import static com.linecorp.armeria.common.HttpMethod.DELETE;
+import static com.linecorp.armeria.common.HttpMethod.GET;
+import static com.linecorp.armeria.common.HttpMethod.HEAD;
+import static com.linecorp.armeria.common.HttpMethod.OPTIONS;
+import static com.linecorp.armeria.common.HttpMethod.PATCH;
+import static com.linecorp.armeria.common.HttpMethod.POST;
+import static com.linecorp.armeria.common.HttpMethod.PUT;
+import static com.linecorp.armeria.common.HttpMethod.TRACE;
+import static com.linecorp.armeria.server.HttpHeaderUtil.ensureUniqueMediaTypes;
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.MediaType;
+
+/**
+ * A builder class for binding a {@link Route} fluently.
+ *
+ * @see DecoratingServiceBindingBuilder
+ * @see ServiceBindingBuilder
+ * @see VirtualHostServiceBindingBuilder
+ */
+class RouteBindingBuilders {
+
+    interface RouteBindingBuilder {
+
+        static RouteBindingBuilder ofOnePath() {
+            return new OnePathRouteBindingBuilder();
+        }
+
+        static RouteBindingBuilder ofMultiPath() {
+            return new MultiPathRouteBindingBuilder();
```

Comment:
Ditto - This is not public API - how about just calling the constructor directly?